### ISSUE TITLE
`no_std` support

### ISSUE THREAD

**Issue by IFcoltransG** [ZetaNumbers/wasm4-rs#9]
I tried to define a `#[panic_handler]` function using this library, but got the following error:
```rs
found duplicate lang item `panic_impl`
the lang item is first defined in crate `std` (which `wasm4` depends on)
first definition in `std` loaded from ...
second definition in the local crate (`cart`)
```
I would like a feature flag to be able to turn off `std` in the `wasm4` crate. I see three occurrences of `std`:
- [`wasm4::utils`](https://github.com/zetanumbers/wasm4-rs/blob/233cb7870c80b14a4a8f5394eab6b0c8e0586b28/wasm4/src/utils.rs#L12) This seems easily replaceable with `core`.
- [`wasm4-impl::include_sprites`](https://github.com/zetanumbers/wasm4-rs/blob/233cb7870c80b14a4a8f5394eab6b0c8e0586b28/wasm4-impl/src/include_sprites.rs#L9) I don't use this in my project, so I imagine it could be disabled by the feature flag. Or if it is only used within the macro at compile time, perhaps it wouldn't even need to be disabled.
- [`wasm4-impl::to_tokens`](https://github.com/zetanumbers/wasm4-rs/blob/233cb7870c80b14a4a8f5394eab6b0c8e0586b28/wasm4-impl/src/include_sprites/to_tokens.rs#L4) I don't use this either.

**Comment by IFcoltransG** [comment]
Having looked closer, it looks like `error` in `core` to fix `utils.rs` is only available under a feature flag `#![feature(error_in_core)]` which would mean this would only be buildable under nightly. So it would be non-trivial to replace it (though it could still be disabled altogether by a feature flag).

**Comment by zetanumbers** [comment]
Right now I am kinda thinking about getting rid of `OutOfDomainError` and wrapper argument types altogether because of having little to no value, but I can gate `std::error::Error` implementation under `std` feature like it's usually done. I think `wasm4-impl` only implements the `include_sprites` proc macro, which executes on the host machine so std is fine if your wasm binary uses `#[no_std]`. And it seems to me that `include_sprites` is also an overengineered and restrictive feature, thus a candidate for move/removal.

However all of those are breaking changes, so I plan to release two new versions, first with cargo features and feature gates, second with major changes like removal of `OutOfDomainError`. As an experiment I'll integrate this crate with my 3D demo written for wasm4.

**Comment by IFcoltransG** [comment]
In the meantime I have discovered a workaround for my use case: `panic::set_hook` can be used to add hooks to the `std` panic, which allows me to run my own code, then use `wasm-snip` to replace the `std`'s panic with `unreachable`.

**Comment by zetanumbers** [comment]
> then use `wasm-snip` to replace the `std`'s panic with `unreachable`.

I've seen something like this with `-Z build-std-features=panic_immediate_abort` in [johnthagen/min-sized-rust](https://github.com/johnthagen/min-sized-rust?tab=readme-ov-file#remove-panic-string-formatting-with-panic_immediate_abort). But I gotta to checkout wasm-snip too.

**Comment by zetanumbers** [comment]
Released the [wasm4-v0.2.0](https://github.com/zetanumbers/wasm4-rs/releases/tag/wasm4-v0.2.0) with no_std support.
